fedimint-core/src/config.rs

```diff
@@ -231,29 +228,42 @@ impl ClientConfig {
 ///
 /// The same `ModuleKind` may have multiple instances with different settings
 #[derive(Debug, Clone, Default)]
-pub struct ConfigGenParams(BTreeMap<String, serde_json::Value>);
+pub struct ConfigGenParams {
+    modules: BTreeMap<ModuleInstanceId, JsonWithKind>,
+    pub registry: ModuleGenRegistry,
+}
 
 impl ConfigGenParams {
     pub fn new() -> ConfigGenParams {
         ConfigGenParams::default()
     }
 
-    /// Add params for a module
+    pub fn generators(&self) -> Vec<(ModuleInstanceId, DynModuleGen)> {
+        self.modules
+            .iter()
+            .map(|(id, kind)| (*id, self.registry.get(kind.kind()).expect("Exists").clone()))
+            .collect()
+    }
+
+    /// Add a module with configuration parameters
     pub fn attach<P: ModuleGenParams>(mut self, module_params: P) -> Self {
-        self.0.insert(
-            P::MODULE_NAME.to_string(),
+        let next_id = self.modules.len() as u16;
```

Comment:
BTW. module instance id don't have to be continuous. In the future, we might allow federation to turn of a certain module and leave it without anything under that ID.

Reply:
Makes sense, ordering them seemed the most simple, plus they need to be consistent with the client for now.